app/main.py

```diff
@@ -1,3 +1,86 @@
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(
+            self,
+            km: int | float
+    ) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(
+            self,
+            other: Union[int, float, "Distance"]
+    ) -> "Distance":
+        if isinstance(other, int | float):
+            return Distance(self.km + other)
+        else:
+            return Distance(self.km + other.km)
+
+    def __iadd__(
+            self,
+            other: Union[int, float, "Distance"]
+    ) -> "Distance":
+        if isinstance(other, int | float):
+            self.km += other
+        else:
+            self.km += other.km
+        return self
+
+    def __mul__(
+            self,
+            other: Union[int, float, "Distance"]
+    ) -> "Distance":
+        return Distance(self.km * other)
+
+    def __truediv__(
+            self,
+            other: Union[int, float, "Distance"]
+    ) -> "Distance":
+        return Distance(round((self.km / other), 2))
```

Comment:
The `__truediv__` method does not handle the case when `other` is a `Distance` object. Additionally, you should handle division by zero to avoid runtime errors.